homeassistant/components/camera/nest.py

```diff
@@ -0,0 +1,107 @@
+"""
+Support for Nest Cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.nest/
+"""
+
+import logging
+from datetime import timedelta
+import requests
+from homeassistant.components.camera import (PLATFORM_SCHEMA, Camera)
+import homeassistant.components.nest as nest
+from homeassistant.util import Throttle
+from homeassistant.util.dt import utcnow
+
+
+DEPENDENCIES = ['nest']
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({})
+
+SIMULATOR_SNAPSHOT_URL = 'https://developer.nest.com/simulator/api/v1/nest/devices/camera/snapshot'
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup a Nest Cam."""
+    if discovery_info is None:
+        return
+    camera_devices = hass.data[nest.DATA_NEST].camera_devices()
+    cameras = [NestCamera(structure, device)
+               for structure, device in camera_devices]
+    add_devices(cameras)
+
+
+class NestCamera(Camera):
+    """Representation of a Nest Camera."""
+
+    def __init__(self, structure, device):
+        """Initialize a Nest Camera."""
+        super().__init__()
+        self.structure = structure
+        self.device = device
+
+        # data attributes
+        self._location = None
+        self._name = None
+        self._is_online = None
+        self._is_streaming = None
+        self._is_video_history_enabled = False
+        # default to non-NestAware subscribed, but will be fixed during update
+        self._time_between_snapshots = timedelta(seconds=30)
+        self._last_image = None
+        self._last_image_at = None
+
+    # FIXME ends up with double name, ie Hallway(Hallway (E5C0))... maybe that's just the simulator?
+    # FIXME duplication with climate/nest
+    @property
+    def name(self):
+        """Return the name of the nest, if any."""
+        return self._name
+        #if self._location is None or self._location == self._name:
+        #    return self._name
+        #else:
+        #    if self._name == '':
+        #        return self._location.capitalize()
+        #    else:
+        #        return self._location.capitalize() + '(' + self._name + ')'
+
+    @property
+    def is_recording(self):
+        """Return true if the device is recording."""
+        return self._is_streaming
+
+    def update(self):
```

Comment:
I mentioned this in https://github.com/home-assistant/home-assistant/pull/4292#discussion_r87705220 that the camera was showing up as a Unnamed Device. I threw an embed in here, and it doesn't actually ever seem to be called.

Is there something else I need to do that the entity is periodically updated?